### ISSUE TITLE
qt5base error while checking hashes

### ISSUE THREAD

**Issue by mlainez** [nerves-project/nerves_system_rpi4#231]
### Environment

* Elixir version (`elixir -v`): 1.15.4
* Nerves environment: main branch
* In our attempt to fix the issues we have with wayland and the drm backend on raspberry pi 4, we are trying alternative options that don't require wayland, such as qt webengine.

### Current behavior

ERROR: while checking hashes from package/qt5/qt5base//qt5base.hash
ERROR: qtbase-da6e958319e95fe564d3b30c931492dd666bfaff.tar.bz2 has wrong sha256 hash:
ERROR: expected: 935d01f5c34903ad9e979431cec7a8a59332ed3fc539e639f5ba87e8d6989b9d
ERROR: got     : 3067c4d84ba9927bfe65bf606c17af082199e0a3b22781fbf9bc6c6bc3de26dd
ERROR: Incomplete download, or man-in-the-middle (MITM) attack

### Expected behavior

I expect it to not fail :)

**Comment by fhunleth** [comment]
I think this might just be a download failure. Could you try erasing `qtbase-da6e958319e95fe564d3b30c931492dd666bfaff.tar.bz2` and then letting Buildroot redownload it? 

As an aside, I'm interested in your Wayland issues. It's been a long time since I've worked with web kiosks, but when I did, I only used Qt. However, since then, I've been told that people have had more success with Wayland. I've since gone the Flutter on Nerves route and that has worked out great, but that obviously changes the front end around quite a bit.

**Comment by fhunleth** [comment]
I'll take a look when I get a moment today.

Regarding where to attribute issues. The Nerves toolchain is very rarely the culprit. It locks a gcc configuration down for both Linux and MacOS rather than using off-the-shelf toolchains which vary subtly. Usually when there's a toolchain "issue", it's because there's a new error or warning that GCC detects that hasn't been fixed. The proper fix is to address it in the package being compiled rather than downgrading the version of GCC you use, since that the error still exists, but just happens to be silent,

Buildroot handles nearly everything else. With the Raspberry Pi, I do modify what Buildroot's choices. Buildroot's maintainers usually pick latest versions for Raspberry Pi-specific packages and don't use Raspberry Pi forks. I lock kernels to tagged versions that were or are distributed with Raspberry Pi OS. For libcamera, I use Raspberry Pi's fork rather than upstream. For the gstreamer error that you're seeing, I'd have to look into it. The issue could be so many things including the Raspberry Pi org creating more forks that need to be referenced or it could just be a config error.

I caught the end of your presentation on the live stream! Someone texted me when you gave it. Super cool!

**Comment by mlainez** [comment]
@fhunleth couldn't find `qtbase-da6e958319e95fe564d3b30c931492dd666bfaff.tar.bz2`  in `~/.nerves/dl/`, so I tried starting from scratch but it did not help.

Thanks for the clarifications regarding the toolchain and others.

I'll create seperate issues for the other problems I am trying to solve just in case someone can help. But I can do without for the moment.

I'm intrigued by the "flutter on nerves" route. I don't mind rewriting the frontend. Do you have any resource I can check to get started or do I look straight into buildroot resources?

**Comment by fhunleth** [comment]
You're right about `qtbase`. The contents of the upstream version changed. I uploaded the one from the Buildroot package cache at https://sources.buildroot.net/qt5base/ to the Nerves package cache. If you run `make` now, you'll get to the next error. 

(I'll get back to you on the other questions next time I get a break)

**Comment by mlainez** [comment]
@fhunleth qtbase goes through now. I'll close the issue since it's resolved.
